File: DemoApp/DemoApp/BaseTableViewController.m
//
//  BaseTableViewController.m
//  DemoApp
//
//  Created by Glispa GmbH on 10.08.16.
//  Copyright © 2016 Glispa GmbH All rights reserved.
//

#import "BaseTableViewController.h"
#import "StreamAdapterTableViewCell.h"
#import "AppControllerManagement.h"

@implementation BaseTableViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self.tableView registerNib:[UINib nibWithNibName:NSStringFromClass([StreamAdapterTableViewCell class]) bundle:[NSBundle mainBundle]] forCellReuseIdentifier:NSStringFromClass([StreamAdapterTableViewCell class])];
}

@end


@interface BaseTableViewController (AppManagement) <AppControllerManagement>
@end

@implementation BaseTableViewController (AppManagement)

- (void)applyUserData:(NSDictionary <NSString *, NSObject *> *)userData {
    if ([userData[@"adUnitId"] isKindOfClass:[NSString class]]) {
        self.adUnitId = (NSString *)userData[@"adUnitId"];
    }
}

@end

File: DemoApp/DemoAppSwift/TableView/BaseTableViewController.swift
//
//  BaseTableViewController.swift
//  DemoApp
//
//  Created by Glispa GmbH on 19/05/16.
//  Copyright © 2016 Glispa GmbH. All rights reserved.
//

import Foundation
import UIKit

@objc(BaseTableViewController)
class BaseTableViewController: UIViewController {
    @IBOutlet weak var tableView: UITableView!
    var adUnitId: String!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        tableView.register(UINib.init(nibName: NSStringFromClass(StreamAdapterTableViewCell.self), bundle: Bundle.main), forCellReuseIdentifier: NSStringFromClass(StreamAdapterTableViewCell.self))
    }
}

extension BaseTableViewController: AppControllerManagement {
    func applyUserData(_ userData: [String:Any]) {
        if let adId = userData["adUnitId"] as? String {
            adUnitId = adId
        }
    }
}
